include/xtensor-io/xhighfive.hpp

```diff
@@ -0,0 +1,666 @@
+    {
+        if ((dmode == xt::dump_mode::create) or (dmode == xt::dump_mode::overwrite and !extensions::exist(file, path)))
+        {
+            return detail::vector::dump_impl<T>::run(file, path, data);
+        }
+
+        return detail::vector::overwrite_impl<T>::run(file, path, data);
+    }
+
+    /**
+    * Write scalar/string to a new DataSet in an open HDF5 file.
+    *
+    * @param file opened HighFive::File (has to be writeable)
+    * @param path path of the DataSet
+    * @param data the data to write
+    * @param dmode DataSet-write mode (xt::dump_mode::create | xt::dump_mode::overwrite)
+    *
+    * @return dataset the newly created HighFive::DataSet (e.g. to add an attribute)
+    */
+    template <class T>
+    inline HighFive::DataSet dump(HighFive::File &file, const std::string &path, const T &data, xt::dump_mode dmode=xt::dump_mode::create)
+    {
+        if ((dmode == xt::dump_mode::create) or (dmode == xt::dump_mode::overwrite and !extensions::exist(file, path)))
+        {
+            return detail::scalar::dump_impl<T>::run(file, path, data);
+        }
+
+        return detail::scalar::overwrite_impl<T>::run(file, path, data);
+    }
+
+    /**
+    * Write a scalar to a rank-one (new, extendible) DataSet in an open HDF5 file.
+    *
+    * @param file opened HighFive::File (has to be writeable)
+    * @param path path of the DataSet
+    * @param data the data to write
+    * @param i the index to which to write
+    *
+    * @return dataset the (newly created) HighFive::DataSet (e.g. to add an attribute)
+    */
+    template <class T>
+    inline HighFive::DataSet dump(HighFive::File &file, const std::string &path, const T &data, std::size_t i)
+    {
+        return detail::scalar::dump_extend_impl<T>::run(file, path, data, i);
+    }
+
+    /**
+    * Write a scalar to a rank-two (new, extendible) DataSet in an open HDF5 file.
+    *
+    * @param file opened HighFive::File (has to be writeable)
+    * @param path path of the DataSet
+    * @param data the data to write
+    * @param i the row to which to write
+    * @param j the column to which to write
+    *
+    * @return dataset the (newly created) HighFive::DataSet (e.g. to add an attribute)
+    */
+    template <class T>
+    inline HighFive::DataSet dump(HighFive::File &file, const std::string &path, const T &data, std::size_t i, std::size_t j)
+    {
+        return detail::scalar::dump_extend_impl<T>::run(file, path, data, i, j);
+    }
+
+    /**
+    * Load entry "(i,j)" from a rank-two DataSet in an open HDF5 file to a scalar.
+    *
+    * @param file opened HighFive::File (has to be writeable)
+    * @param i the row to which to load
+    * @param j the column to which to load
+    * @param path path of the DataSet
+    *
+    * @return data the read data
+    */
+    template <class T>
+    inline auto load(const HighFive::File &file, const std::string &path, std::size_t i, std::size_t j)
```

Comment:
I am wondering if having this scalar interface is a good idea? Since it's arbitrarily limited to 1- and 2D...

There are two options that I see:

- create a load_scalar function that takes n-long index type
- create a fake index type that inherits from some container, that can be used in the function, e.g. `load(file, path, h5idx{4, 1, 2})`
- Or we can make another function called something like `load_inplace` that doesn't return a copy of the data but rather wraps the pointer to the data set with `xt::adapt` - the user could then index into that to extract a scalar.

Reply:
I see the point. What about `load<double>(file, path, {4, 1, 2})`? The indices can then be a `std::vector`, which is what they are upstream as well. I'm not a big fan of creating a new function name such as `load_scalar`.

With respect to you suggestion. That is obviously the coolest option! However, I cannot estimate the investment we would have to make.

Reply:
Ok, then let's modify the existing function to take an index first.
I might make a PR later to add this adapt functionality (we could even use mmap for really large datasets). You'll get to review it :)